### ISSUE TITLE
GNOME autologin support

### ISSUE THREAD

**Issue by samcday** [samcday/phrog#2]
How does it work? Is it GDM specific?

**Comment by samcday** [comment]
Related: https://gitlab.com/mobian1/phog/-/issues/14

**Comment by agx** [comment]
You're likely looking for something else but this is how you can auto login with greetd: https://salsa.debian.org/DebianOnMobile-team/phog/-/commit/9501a29e7639eec21e0f4ab074116e0b215ce849

**Comment by samcday** [comment]
Yeah, I was thinking more about how to pump the config from GNOME Settings to make it more seamless.

This issue is somewhat related to #5 though... I've realized that greetd might be a little *too* simplistic. It only takes over one VTY, and when a session is chosen by the greeter, the greeter is killed and the actual session is spawned in that same VTY.

This means no multi-session/seat support, and also no proper GNOME session locking support (gnome session just wants to do a D-Bus thing and let the display manager take care of it). I think it also guarantees that there's no reliable way to seamlessly (without modesetting blanking/artifacting or brief flashes of console output) transition from greeter -> session.

**Comment by samcday** [comment]
> I'm likely missing something but session locking should be independent as long as loginctl lock-session works (which it does with phosh).

Much more likely that **I'm** the one missing something :sweat_smile:  The GNOME session shows me a warning that locking is unsupported when I spawn it on Fedora via phrog+greetd.

> I somewhat considered that simplicity a feature but yes

Yes, my initial reaction when I was considering this was similar. However, I think a lot of what makes greetd "simple" is that it handles *only* the VT+PAM interactions and delegates the rest to IPC. So maybe what I'm thinking of could still fit in the greetd ethos, or maybe it's possible to layer something on top of greetd, or fork and support the same IPC mechanism for interop with existing greeters.

**Comment by agx** [comment]
> The GNOME session shows me a warning that locking is unsupported when I spawn it on Fedora via phrog+greetd.

Can you share the exact message?

**Comment by samcday** [comment]
Sure:

![image](https://github.com/user-attachments/assets/a9378b52-d76f-408c-9dc2-cf8df784f0a2)

In addition, the logs contain the message I posted in the description of #5 

I was trying to figure out how to make that shy message re-appear in order to screenshot it for you. That led me to https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/2432 where this behaviour is spoken about as fact.

**Comment by agx** [comment]
Found that in the sources now, thanks. (The one in #5 comes from g-s-d which looks like a symptom of gnome-shell refusing to lock). I would understand why it's needed for session switching but I currently don't get why it would be needed for locking the session. Would be interesting to see if one can mock enough of GDMs DBus interface to get that going.

**Comment by samcday** [comment]
Well the thing is, I wonder if gnome-session even has its own "unlock" page? I'm not particularly comfortable in the GNOME sources so maybe you know the answer to this?

That is, I wouldn't be surprised if the choice was made at some point to completely remove any form of lockscreen from gnome-session and it instead only resides in GDM, hence the enforced requirement.

**Comment by agx** [comment]
gnome-shell has the unlock page (and it's also what sends the notification). AFAIK gnome-session isn't involved.

**Comment by samcday** [comment]
Ah, for some reason I had equated `gnome-shell` == `gnome-session` in my head :sweat_smile:

**Comment by samcday** [comment]
So perhaps it will only require a modest amount of `org.gnome.DisplayManager` shimming to convince gnome-shell that it is safe to do lock things ;)

**Comment by samcday** [comment]
As of very recently I'm back to daily-driving :frog: on my desktop + laptop where I'm using GNOME shell. Not being able to lock is something I can live with for the moment but I think it will irritate me enough that I'll sit down and look at this properly soon enough :sweat_smile:

**Comment by samcday** [comment]
This is not at all feasible at current, I think.

[It turns out that AccountsService is manually parsing out the symlink target of `/etc/systemd/system/display-manager.service` in order to determine which DM is active.](https://gitlab.freedesktop.org/accountsservice/accountsservice/-/blob/f84a77f4/src/daemon.c#L208-217)

*Maybe* it's worth adding greetd support there? I don't really know how that would work though, since the configuration for greetd "autologin" requires knowing the gritty information about which process to start as the initial session. :thinking: 

As it currently stands, AccountsService is sorta a mess. On my F40 machine the `23.13.9` version of AccountsService just segfaults when toggling automatic login on a user in GNOME Settings. I think that's fixed in newer versions so maybe my F41 will just grey out the option entirely or something if an unsupported DM is detected? :shrug:

**Comment by samcday** [comment]
This work would require changes in greetd and AccountsService which I am not interested in pursuing.

**Comment by samcday** [comment]
Issue, I summon thee from beyond the grave. Heed my call!
